src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -231,7 +244,15 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
    * @returns {Array.<Object>}
    */
   getSegments(_up : number, _to : number) : ISegment[] {
-    return getSegmentsFromTimeline(this._index, _up, _to, this._scaledPeriodEnd);
+    return getSegmentsFromTimeline(this._index, _up, _to, this._scaledPeriodEnd)
```

Comment:
As `getSegmentsFromTimeline` is defined only for DASH RepresentationIndexes can't we just provide `isInbandEventWhitelisted` to that function instead of repeating that code at each call site?